macros/sql/star.sql

```diff
@@ -5,14 +5,8 @@
         {{ return('') }}
     {% endif %}
 
-    {%- if from.name -%}
-        {%- set schema_name, table_name = from.schema, from.name -%}
-    {%- else -%}
-        {%- set schema_name, table_name = (from | string).split(".") -%}
-    {%- endif -%}
-
     {%- set include_cols = [] %}
-    {%- set cols = adapter.get_columns_in_table(schema_name, table_name) -%}
+    {%- set cols = adapter.get_columns_in_relation(from) -%}
```

Comment:
I think there might be merit to adding a check here that returns an error if a non-relation is provided to this macro (and others). This would have the added benefit of raising an error if someone passes an ephemeral model into `star`. 

Maybe:
```
{% if from.get('name') is none %}
  {% do exceptions.raise_compiler_error("Macro star() expected a Relation but received the value: " ~ from) %}
{% endif %}
```

or something. We could wrap this up in its own little macro as a helper :)

Reply:
Good idea! We might add this error to all of the macros that now expect relation arguments in a break-y sort of way. Two q's:
    * Is there a better way to infer the argument type, or is `arg.get('name') != none` the best way to confirm that `arg` is a relation?
    * Is there a Jinja method you know of that lets us pass the name of the current macro? Relevant if we were to make `error_if_not_relation(arg)` its own macro, and wanted to pretty print a dynamic compiler error `"Macro " ~ my_macro ~ " expected a Relation for argument " ~ arg"`

Reply:
These are really good questions!

I don't think we have a great mechanism in place to assert the types of objects from the jinja context. We could conceivably add a context function or a jinja test (like `abc is Relation`) but that doesn't exist today :/. I think our best option is going to be to look for a `name` attribute. Here's some example code I played around with:

```
{% set things = [
    'abc.def',
    api.Relation.create(schema='ghi', identifier='jkl'),
] %}


{% for thing in things %}
    {{ log(thing ~ " is relation? " ~ (thing is mapping and thing.name is defined), info=true) }}
{% endfor %}
```

Let's be sure to test this with https://github.com/fishtown-analytics/dbt/pull/1416

We also don't have a way to get the macro name from the macro context. I think we should just anticipate making a macro to handle raising the exception, then pass in a string to use as the macro name from the caller.

Reply:
Right on. Is an `is_relation()` macro something that I should add to utils now, or do you anticipate creating it within the context of dbt proper?

Reply:
Yeah - let's add it to dbt-utils! Maybe name it something like `_is_relation()` since I wouldn't be comfortable with folks using this externally (yet). If we had some way of locally scoping the macro I'd suggest that, but we can just indicate its private-ness with a naming convention for now.

Reply:
Possibly related to the PR you posted: `from.name` _does_ return a value, but `from.get('name')` _does not_. The macro fails the Relation test as written above.

Reply:
Hmm, yeah, we'll probably want to call `from.name` then. @beckjake off the top of your head, can you think of a better way to tell if an object is a Relation in the jinja context than `obj is mapping and obj.name is not none`?

Reply:
Yeah, `.name` is a property that actually resolves to `.path.get('identifier')`.

I think I'd do ``obj is mapping and 'type' in obj.get('metadata', {}))`. We should probably add a jinja test for this, or something.

Reply:
Ew, maybe you have to even check the type string:

`obj is mapping and obj.get('metadata', {}).get('type', '').endswith('Relation')`

Reply:
Implemented the most rigorous logic in a macro `_is_incremental()`, which returns the following exception:
```
Compilation Error
  Macro star expected a Relation but received the value: hello

  > in macro _is_relation (macros/cross_db_utils/_is_relation.sql)
  > called by macro star (macros/sql/star.sql)
  > called by <Unknown>
```